core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java

```diff
@@ -1654,6 +1675,19 @@ private void checkLiteral2(String expression, String expected) {
     sql(query).ok(expected);
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-3112">[CALCITE-3112]
+   * Support Window in RelToSqlConverter</a>. */
+  @Test public void testLeadFunctionForPrintingOfFrameBoundaryWithWindow() {
+    String query = "SELECT lead(\"employee_id\",1,'NA') over "
+            + "(partition by \"hire_date\" order by \"employee_id\") FROM \"employee\"";
+    String expected = "SELECT LEAD(\"employee_id\", 1, 'NA') OVER "
+            + "(PARTITION BY \"hire_date\" "
+            + "ORDER BY \"employee_id\") AS \"$0\"\n"
+            + "FROM \"foodmart\".\"employee\"";
+    sql(query).includeWindow(true).ok(expected);
+  }
+
```

Comment:
You can put these test cases together, so that you don't need to repeat the comments and JIRA link many times.

Reply:
Thanks, I have addressed.